solution/src/integrationTest/kotlin/com/cosmotech/solution/service/SolutionServiceRBACTest.kt

```diff
@@ -0,0 +1,1074 @@
+              every { getCurrentAccountIdentifier(any()) } returns CONNECTED_ADMIN_USER
+
+              val organization = mockOrganization()
+              val solution = mockSolution(organization.id!!, role = role)
+
+              organizationSaved = organizationApiService.registerOrganization(organization)
+              solutionSaved = solutionApiService.createSolution(organizationSaved.id!!, solution)
+
+              ReflectionTestUtils.setField(solutionApiService, "resourceScanner", resourceScanner)
+              every { resourceScanner.scanMimeTypes(any(), any()) } returns Unit
+              mockkStatic(ArchiveStreamFactory::class)
+              every { ArchiveStreamFactory.detect(any()) } returns "zip"
+              ReflectionTestUtils.setField(
+                  solutionApiService,
+                  "azureStorageBlobServiceClient",
+                  azureStorageBlobServiceClient)
+              every {
+                azureStorageBlobServiceClient
+                    .getBlobContainerClient(any())
+                    .getBlobClient(any())
+                    .upload(any(), any(), any())
+              } returns Unit
+
+              every { resource.contentLength() } returns 0
+              every { resource.getInputStream() } returns inputStream
+              every { inputStream.read() } returns 0
+
+              val runTemplateHandlerId = RunTemplateHandlerId.values().first()
+
+              every { getCurrentAccountIdentifier(any()) } returns FAKE_MAIL
+
+              if (shouldThrow) {
+                val exception =
+                    assertThrows<CsmAccessForbiddenException> {
+                      solutionApiService.uploadRunTemplateHandler(
+                          organizationSaved.id!!,
+                          solutionSaved.id!!,
+                          "runTemplate",
+                          runTemplateHandlerId,
+                          resource,
+                          true)
+                    }
+                if (role == ROLE_NONE) {
+                  assertEquals(
+                      "RBAC ${solutionSaved.id!!} - User does not have permission $PERMISSION_READ",
+                      exception.message)
+                } else {
+                  assertEquals(
+                      "RBAC ${solutionSaved.id!!} - User does not have permission $PERMISSION_WRITE",
+                      exception.message)
+                }
+              } else {
+                assertDoesNotThrow {
+                  solutionApiService.uploadRunTemplateHandler(
+                      organizationSaved.id!!,
+                      solutionSaved.id!!,
+                      "runTemplate",
+                      runTemplateHandlerId,
+                      resource,
+                      true)
+                }
+              }
+            }
+          }
+
+  fun mockOrganization(
+      id: String = "organization_id",
+      userName: String = FAKE_MAIL,
+      role: String = ROLE_ADMIN
```

Comment:
should not have default value. this force to explicit what user is used and what role is given to it.